src/cpp/shared_core/include/shared_core/FilePath.hpp

```diff
@@ -231,6 +260,53 @@ class FilePath
     */
    static Error uniqueFilePath(const std::string& in_basePath, const std::string& in_extension, FilePath& out_filePath);
 
+#ifndef _WIN32
+   /**
+    * @brief Changes the file mode to the specified file mode.
+    *
+    * @param in_fileModeStr     The posix file mode string. e.g. rwxr-xr-x.
```

Comment:
Is it worth documenting that only a subset of posix file mode strings are supported?

Reply:
Yes, that seems worth it to me. I'm not sure which are supported and which aren't off-hand. Do you know where I can find that information? Or, if you happen to know feel free to suggest an edit.

Reply:
Wait, I think I may have misunderstood -- I was looking at the code in https://github.com/rstudio/rstudio/blob/91d3851f1248b40da3fafa0ed4d15bcacd6e50b2/src/cpp/core/include/core/system/FileMode.hpp#L131-L175 and thought that denoted the "supported" file modes, but that appears to be something separate. Sorry!

Reply:
Oh okay!